providers/mlx5/mlx5.c

```diff
@@ -305,14 +305,14 @@ man cpuset
 */
 static void mlx5_local_cpu_set(struct ibv_device *ibdev, cpu_set_t *cpu_set)
 {
-	char *p, buf[1024];
+	char *p, *buf = NULL;
```

Comment:
This is used in the 'else' flow below (317) as a buffer, the code here is wrong.